pkg/aws/ec2_helpers_test.go

```diff
@@ -175,5 +179,248 @@ var _ = Describe("AWS EC2 Helper Functions", func() {
 			Entry("Negative test - not an IP address", "Not an IP address", true),
 		)
 	})
+	// A unit test for configureInstance. Bypasses CreateEc2CloudConfig's logic to isolate cases where its logic may be
+	// faulty, hence the helper function newDefaultValidEC2ConfigForInstance which provides a neutral all-good
+	// AWSEc2DynamicConfig with baseline and passing generic configurations.
+	// Breaks the various configuration fields to families of configuration topics, each tested in it own Context. Some
+	// have happy test paths, some have sad ones.
+	Describe("Testing configureInstance", func() {
+		var (
+			ecConfig       AWSEc2DynamicConfig
+			taskRunName    string
+			instanceTag    string
+			additionalTags map[string]string
+		)
+
+		BeforeEach(func() {
+			taskRunName = "cfg-taskrun"
+			instanceTag = "cfg-instance"
+			additionalTags = make(map[string]string)
+			ecConfig = newDefaultValidEC2ConfigForInstance()
+		})
+
+		Context("when configuring basic instance details", func() {
+			It("should correctly set KeyName, AMI, and InstanceType", func() {
+				runInput := ecConfig.configureInstance(taskRunName, instanceTag, additionalTags)
+				Expect(runInput.KeyName).To(PointTo(Equal(ecConfig.KeyName)))
+				Expect(runInput.ImageId).To(PointTo(Equal(ecConfig.Ami)))
+				Expect(runInput.InstanceType).To(Equal(types.InstanceType(ecConfig.InstanceType)))
+			})
+
+			It("should set MinCount=1, MaxCount=1, EbsOptimized=true, and ShutdownBehavior=terminate", func() {
+				runInput := ecConfig.configureInstance(taskRunName, instanceTag, additionalTags)
+				Expect(runInput.MinCount).To(PointTo(Equal(int32(1))))
+				Expect(runInput.MaxCount).To(PointTo(Equal(int32(1))))
+				Expect(runInput.EbsOptimized).To(PointTo(BeTrue()))
+				Expect(runInput.InstanceInitiatedShutdownBehavior).To(Equal(types.ShutdownBehaviorTerminate))
+			})
+		})
+
+		Context("when configuring networking", func() {
+			DescribeTable("with various subnet and security group settings",
+				// This DescribeTable takes an ecConfig-mutating function and a function that verifies the new content
+				// of the ecConfig after going through configureInstance
+				func(setupConfig func(conf *AWSEc2DynamicConfig), verify func(input *ec2.RunInstancesInput)) {
+					currentConfig := ecConfig
+					setupConfig(&currentConfig)
+					runInput := currentConfig.configureInstance(taskRunName, instanceTag, additionalTags)
+					verify(runInput)
+				},
+				Entry("using baseline SubnetId and SecurityGroupId",
+					func(conf *AWSEc2DynamicConfig) {
+						// ecConfig is as it's created in BeforeEach.
+					},
+					func(input *ec2.RunInstancesInput) {
+						Expect(input.SubnetId).To(PointTo(Equal(ecConfig.SubnetId)))
```

Comment:
AFAYK, if there's a way of removing most of these `PointTo`?

Reply:
I went over these following Andy's suggestion, and found only one place where we can replace it with something that's different and 100% does the same job.
I looked into Gomega's  `BeComparableTo` but it could only replace some fields that aren't handled by aws.String() or aws.Int32() (which create a brand new pointer). Look at the commit I put in following Andy's suggestion, it is the most I saw I could do.

Reply:
This one could be taken out:
```suggestion
						Expect(input.SubnetId).To(Equal(&ecConfig.SubnetId))
```

gomega's `Equal` matcher doesn't care if pointers are equal, it looks at the underlying object and uses `reflect.DeepEqual` to test for equality.  If the intention is to test for pointer equality, you may want to use [BeIdenticalTo](https://onsi.github.io/gomega/#beidenticaltoexpected-any) instead.